safe-global/safe-wallet-web
# Refactor: onboard auto connect

## What it solves

Resolves #2306 and #2342.

Onboard has its own auto-connect-to-last-wallet feature. This PR removes our old custom logic and uses that.

## How to test this
 * Make sure the same wallet re-connects when reloading the page
 * Try locking MetaMask manually and it shouldn't re-connect or even popup MetaMask after a page reload
<img width="257" alt="Screenshot 2023-08-03 at 13 46 47" src="https://github.com/safe-global/safe-wallet-web/assets/381895/91ac4729-dcac-4b9e-898f-a3e84955f6b8">

 * Check that we still track wallet connects (both manual and automatic reconnections)
 * Check that we now properly track which wallet is used with WalletConnect
